src/microbe_stage/BioProcess.cs

```diff
@@ -64,6 +64,25 @@ public void Check(string name)
         TranslationHelper.CopyTranslateTemplatesToTranslateSource(this);
     }
 
+    public float GetRateWithConditions(BiomeConditions conditions)
+    {
+        var rate = 1.0f;
+
+        foreach (var input in Inputs)
+        {
+            if (input.Key.IsEnvironmental)
+            {
+                foreach (var compound in conditions.AverageCompounds)
+                {
+                    if (compound.Key.Equals(input.Key) && rate > 0)
```

Comment:
I think that this can be simplified by getting compound with `conditions.AverageCompounds.TryGetValue()` instead of `foreach`, as the result should be, as I understand, the same, but with less calculations